### PR TITLE
Add webmoney payment option

### PR DESCRIPTION
**Pull request by rramsden**
This pull request adds the Webmoney payment option to Spree.

A few notes:

* Multi-card transactions in Komoju require PATCH instead of POST'ing to a resource. ActiveMerchant supports PATCH in the latest release, however, Spree 3.0.4 version locks ActiveMerchant @ 1.47.0. Because of this I've created a extension to `connection.rb` which is only required with ActiveMerchant == '1.47.0' which adds `:patch`
* I ended up using Javascript instead of Coffescript. For some reason `.js.coffee` files were not being recognised. I think this is because assets are compiled in the `vendor` directory rather that into `app/assets`.

image

### CODE REVIEW

**Comment by rramsden** [comment]
```
["camirb(main):001:0> ["camelmasa", "shioyama"].sample
=> "camelmasa"
```

@camelmasa Could you review :bow: :pray:

File: spec/models/web_money_decorator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      source = Spree::WebMoney.new(prepaid_cards: [], short_amount: 1000, payment_uuid: "123")
+      decorator = WebMoneyDecorator.new(source)
+
+      expect(decorator.prepaid_cards).to eq([])
+      expect(decorator.payment_uuid).to eq("123")
+      expect(decorator.short_amount).to eq(1000)
+    end
+  end
+
+  context 'when source is nill' do
```
Comment (camelmasa) [inline spec/models/web_money_decorator_spec.rb]: typo `nil` ? 🐜

Comment (rramsden) [inline spec/models/web_money_decorator_spec.rb]: No, its when the constructor is passed `nil`

File: app/models/spree/gateway/komoju_web_money.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      response
+    end
+
+    def continue_uuid(current_order_id)
+      order = Spree::Order.find_by_number(current_order_id.split('-').first)
+      
+      # NOTE: Since the current payment is this payment, find the payment before
+      # and see if it was for a webmoney transaction. If it was check if it has a payment UUID
+      # to continue the transaction.
+      source = WebMoneyDecorator.new(order.payments.last(2).first.try(:source))
```
Comment (camelmasa) [inline app/models/spree/gateway/komoju_web_money.rb]: Note: You said we can remove this line.

We can do following.
```rb
order.payments.last(2).first.try(:source).try(:payment_uuid)
```

File: app/models/web_money_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class WebMoneyDecorator
```
Comment (camelmasa) [inline app/models/web_money_decorator.rb]: I think this class name needs namespace for avoid conflict :santa: 

e.g. `SpreeKomoju::WebmoneyDecorator`.

Comment (rramsden) [inline app/models/web_money_decorator.rb]: Good idea!

**Comment by camelmasa** [comment]
I checked It works that :smiley: 

spree_demo_site

I'll add translation for japanese after merged it.

spree_demo_site

**Comment by camelmasa** [comment]
I needed to select WebMoney radio button everytime on payment page when insufficient funds error.

If possible, I think customers prefers that the radio button is selected WebMoney when insufficient funds error :wink:

**Comment by rramsden** [comment]
@camelmasa Yes, will need some javascript to select the payment option again. This is not supported by Spree.

**Comment by camelmasa** [comment]
@rramsden no problem :+1:

**Comment by rramsden** [comment]
@camelmasa I've namespaced it under Spree which I think should be fine. `SpreeKomoju` defines the engine, extensions, and exceptions.

**Comment by camelmasa** [comment]
@rramsden LGTM :sta

**Pull request merged**